### PR TITLE
Bret Funk 1705 Mod 1

### PR DESCRIPTION
**Pull request by brettt**

### CODE REVIEW

File: lib/pantry.rb [hunk 1]
```diff
@@ -1,3 +1,82 @@
...
+  end
+
+  def stock_check(item)
+    stock[item].nil? ? 0 : stock[item]
+  end
+
+  def restock(item, quantity)
+    stock[item] ||= 0
+    stock[item] += quantity
+  end
```
Comment (s-espinosa) [inline lib/pantry.rb]: This is super clean. Nice use of memoization.

File: lib/pantry.rb [hunk 2]
```diff
@@ -1,3 +1,82 @@
...
+
+  def convert_units(recipe)
+    recipe.ingredients.each_pair do |item, quantity|
+      recipe.ingredients[item] = {
+        quantity: correct_quantity(quantity),
+        units:    correct_units(quantity)
+      }
+    end
+  end
```
Comment (s-espinosa) [inline lib/pantry.rb]: Reading the spec I would expect this to *return* the new hash, not change the hash on the instance of recipe.

I can see the logic here, but with clients/employers you're going to want to build things to spec absent a conversation to discuss.

File: lib/pantry.rb [hunk 3]
```diff
@@ -1,3 +1,82 @@
...
+
+  #for iteration four, work in progress
```
Comment (s-espinosa) [inline lib/pantry.rb]: Normally would tell you to make sure to take the comments out, but thanks for these notes.

File: lib/pantry.rb [hunk 4]
```diff
@@ -1,3 +1,82 @@
...
+    if ingredients.quantity[1] == "Milli-Units"
+      quantity * 1000
+    elsif ingredients..quantity[1] == "Universal Units"
+      quantity * 1
+    elsif ingredients..quantity[1] == "Centi-Units"
+      quantity / 100
+    end
+  end
+
+  def correct_quantity(quantity)
```
Comment (s-espinosa) [inline lib/pantry.rb]: Really like how you broke these two methods out.

File: lib/pantry.rb [hunk 5]
```diff
@@ -1,3 +1,82 @@
...
+      "Milli-Units"
+    elsif quantity > 1 && quantity < 100
+      "Universal Units"
+    elsif quantity > 100
+      "Centi-Units"
+    end
+  end
+
+#not adding values for some dumb reason
+  def add_to_shopping_list(recipe)
```
Comment (s-espinosa) [inline lib/pantry.rb]: Try with an `@`.

File: test/pantry_test.rb [hunk 6]
```diff
@@ -4,4 +4,229 @@
...
+
+  def test_empty_stock
+    result = pantry.stock
+    pantry_hash = {}
+
+    assert_equal pantry_hash, result
+  end
+
+  def test_stock_check_soylant
+    result = pantry.stock_check("Soylant")
```
Comment (s-espinosa) [inline test/pantry_test.rb]: Ha!

File: test/pantry_test.rb [hunk 7]
```diff
@@ -4,4 +4,229 @@
...
+  def test_recipe_format_many_items
+    recipe.add_ingredient("Flour", 500)
+    recipe.add_ingredient("Cheese", 200)
+    recipe.add_ingredient("Love", 1500)
+    ingredients = {"Flour"=>500, "Cheese"=>200, "Love"=>1500}
+
+    assert_equal ingredients, recipe.ingredients
+  end
+
+  def test_correct_units_milli
```
Comment (s-espinosa) [inline test/pantry_test.rb]: Like the trio of tests below.

File: test/pantry_test.rb [hunk 8]
```diff
@@ -4,4 +4,229 @@
...
+    r.add_ingredient("Cheese", 5)
+    r.add_ingredient("Flour", 20)
+    r.add_ingredient("Milk", 10)
+    pantry.add_to_shopping_list(r)
+    result = pantry.print_shopping_list
+
+    assert_equal 3, result.length
+  end
+
+  def test_new_convert_units_cayenne
```
Comment (s-espinosa) [inline test/pantry_test.rb]: Interesting! Did you write all of your tests first? I would probably recommend using a shorter test/implement cycle.

**Comment by brettt** [comment]
Thanks for the feedback.  In case your questions weren't retorical here are my answers:

For the comment about changing the original hash, i wanted to use map but i thought to iteratate over a hash and use both values you needed to use each_pair but your response tells me I am wrong (map_pair?).

Comments were used on iteration 4 methods because I was still on red-to-green and didn't want to get dinged for sloppy code (unless you would rather I delete work in progress).

Thanks for you feedback!  This was super fun.

_____________________________

**Pull request closed without merging**